ee/tuf/autoupdate.go

```diff
@@ -44,10 +46,41 @@ const (
 
 var binaries = []autoupdatableBinary{binaryLauncher, binaryOsqueryd}
 
+func (a *autoupdatableBinary) UnmarshalJSON(data []byte) (err error) {
+	var binary string
+	if err := json.Unmarshal(data, &binary); err != nil {
+		return err
+	}
+	switch binary {
+	case "launcher":
+		*a = binaryLauncher
+	case "osqueryd":
+		*a = binaryOsqueryd
+	default:
+		return fmt.Errorf("unknown binary %s", binary)
```

Comment:
Will this create an upgrade issue? like if launcher v3 wants to upgrade `launcher`, `osquery`, and `foobar`, then all the old launchers will fail to unmarshal

Reply:
Oh fair -- I will update this